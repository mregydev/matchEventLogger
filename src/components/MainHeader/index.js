import React from "react";
import { Grid, Tooltip } from "@material-ui/core";
import { SportsSoccer, GitHub } from "@material-ui/icons";
import Messages from "../../Messages";
import styles from "./style.module.scss";

//Functional Component that draws the main application header
export default () => {
  return (
    <Grid container className={styles.appHeader}>
      <Grid item xs={12} className={styles.headerContent}>
        <Grid item xs={10}>
          <span className={styles.headerLogo}>
            <SportsSoccer />
          </span>
          <span> {Messages.AppHeader}</span>
        </Grid>
        <Grid item xs={2}>
          <Tooltip title={Messages.GithubLink}>
            <a
              href="https://github.com/mregydev/matchEventLogger"
              className={styles.githubLink}
            >
              <GitHub />
            </a>
          </Tooltip>
        </Grid>
      </Grid>
    </Grid>
  );
};
