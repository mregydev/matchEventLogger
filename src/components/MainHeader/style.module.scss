@import "../../styles/variables";

.appHeader {
  background: $mainColor;
  color: white;
  padding: 10px;
  font-weight: bold;
  height: 8vh;
  font-size: $headerFontSize;
  font-family: sans-serif;
}

.headerContent {
  display: flex;
  align-items: center;
}

.headerLogo {
  margin-right: 10px;
}

.githubLink {
  color: white;
  font-weight: bold;
  float: right;
  
}
