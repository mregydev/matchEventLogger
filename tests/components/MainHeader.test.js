import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MainHeader from "../../src/components/MainHeader";

describe("<MainHeader /> test cases", () => {
  test("renders", () => {
    const wrapper = shallow(<MainHeader />);
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
