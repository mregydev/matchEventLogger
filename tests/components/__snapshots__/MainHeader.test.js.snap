// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainHeader /> test cases renders 1`] = `""`;
